app/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/returns/address/MixedUseController.scala

```diff
@@ -0,0 +1,429 @@
+  protected lazy val backLinkCall: EnteringSingleMixedUsePropertyDetails => Call = _ => enterPostcodeCall
+  protected lazy val ukAddressNotAllowedExitPageCall: Option[Call]               = None
+
+  protected lazy val enterUkAddressCall: Call       = addressRoutes.MixedUseController.enterUkAddress()
+  protected lazy val enterUkAddressSubmitCall: Call = addressRoutes.MixedUseController.enterUkAddressSubmit()
+  protected lazy val enterPostcodeCall: Call        = addressRoutes.MixedUseController.enterPostcode()
+  protected lazy val enterPostcodeSubmitCall: Call  = addressRoutes.MixedUseController.enterPostcodeSubmit()
+  protected lazy val selectAddressCall: Call        = addressRoutes.MixedUseController.enterUkAddress()
+  protected lazy val selectAddressSubmitCall: Call  = addressRoutes.MixedUseController.selectAddress()
+  protected lazy val continueCall: Call             = addressRoutes.MixedUseController.checkYourAnswers()
+
+  def singleMixedUseGuidance(): Action[AnyContent] =
+    authenticatedActionWithSessionData.async { implicit request =>
+      withValidJourney(request) { (_, r) =>
+        val backLink =
+          r.draftReturn.mixedUsePropertyDetailsAnswers
+            .getOrElse(IncompleteMixedUsePropertyDetailsAnswers.empty)
+            .fold(
+              _ => controllers.returns.routes.TaskListController.taskList(),
+              _ => routes.MixedUseController.checkYourAnswers()
+            )
+        Ok(
+          singleMixedUseGuidance(
+            backLink
+          )
+        )
+      }
+    }
+
+  def singleMixedUseGuidanceSubmit(): Action[AnyContent] =
+    authenticatedActionWithSessionData.async { implicit request =>
+      withValidJourney(request) { (_, _) =>
+        Redirect(enterPostcodeCall)
+      }
+    }
+
+  def enterDisposalValue(): Action[AnyContent] =
+    authenticatedActionWithSessionData.async { implicit request =>
+      withValidJourney(request) { (_, r) =>
+        val answers       = r.draftReturn.mixedUsePropertyDetailsAnswers
+          .getOrElse(IncompleteMixedUsePropertyDetailsAnswers.empty)
+        val backLink      = routes.MixedUseController.singleMixedUseGuidance()
+        val disposalPrice = answers.fold(_.disposalPrice, c => Some(c.disposalPrice))
+        val form          = disposalPrice.fold(disposalPriceForm)(c => disposalPriceForm.fill(c.inPounds))
+        Ok(
+          disposalValuePage(
+            form,
+            backLink,
+            r.journey.subscribedDetails.isATrust,
+            r.representativeType
+          )
+        )
+      }
+    }
+
+  def enterDisposalValueSubmit(): Action[AnyContent] =
+    authenticatedActionWithSessionData.async { implicit request =>
+      withValidJourney(request) { (_, r) =>
+        val answers  = r.draftReturn.mixedUsePropertyDetailsAnswers
+          .getOrElse(IncompleteMixedUsePropertyDetailsAnswers.empty)
+        val backLink = routes.MixedUseController.singleMixedUseGuidance()
```

Comment:
is this the correct back link? It makes more sense for it to go to the enter postcode page. Also, if the answers are complete then the back link should point to the cya page